src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -319,5 +364,56 @@ private static string GetBucket(AugmentedTimelineIssue augIssue)
         {
             return await azureServer.ListBuilds(project, cancellationToken, minDateTime, limit);
         }
+
+        private async Task GetImageNames(List<AugmentedTimelineRecord> records, CancellationToken cancellationToken)
+        {
+            SemaphoreSlim throttleSemaphore = new SemaphoreSlim(50);
+
+            var bag = new ConcurrentBag<Task>();
+            int canceledTasksNumber = 0;
+
+            foreach(var record in records)
+            {
+                if(!string.IsNullOrEmpty(record.Raw.Log?.Url) && record.Raw.Name == "Initialize job")
+                {
+                    var childTask = Task.Run(async () =>
+                    {
+                        try
+                        {
+                            await throttleSemaphore.WaitAsync();
```

Comment:
The wait should go before the try.  If the Wait fails, it's important that "Release" not be called.

Reply:
moved